test/Microsoft.Azure.ServiceBus.UnitTests/ExpectedMessagingExceptionTests.cs

```diff
@@ -72,58 +72,55 @@ public class ExpectedMessagingExceptionTests
         {
-            var messagingFactory = new ServiceBusClientFactory();
-            var queueClient =
-                (QueueClient)messagingFactory.CreateQueueClientFromConnectionString(
-                    TestUtility.GetEntityConnectionString(TestConstants.SessionNonPartitionedQueueName));
+            var sender = new MessageSender(TestUtility.NamespaceConnectionString, TestConstants.SessionNonPartitionedQueueName);
+            var sessionClient = new SessionClient(TestUtility.NamespaceConnectionString, TestConstants.SessionNonPartitionedQueueName);
 
             try
             {
                 var messageId = "test-message1";
                 var sessionId = Guid.NewGuid().ToString();
-            await queueClient.SendAsync(new Message
-                { MessageId = messageId, SessionId = sessionId });
+                await sender.SendAsync(new Message() { MessageId = messageId, SessionId = sessionId });
                 TestUtility.Log($"Sent Message: {messageId} to Session: {sessionId}");
 
-                MessageSession messageSession = await queueClient.AcceptMessageSessionAsync(sessionId);
-            Assert.NotNull(messageSession);
+                var sessionReceiver = await sessionClient.AcceptMessageSessionAsync(sessionId);
+                Assert.NotNull(sessionReceiver);
+                TestUtility.Log($"Received Session: SessionId: {sessionReceiver.SessionId}: LockedUntilUtc: {sessionReceiver.LockedUntilUtc}");
 
-                var message = await messageSession.ReceiveAsync();
+                Message message = await sessionReceiver.ReceiveAsync();
                 Assert.True(message.MessageId == messageId);
-                TestUtility.Log($"Received Message: SessionId: {messageSession.SessionId}");
+                TestUtility.Log($"Received Message: MessageId: {message.MessageId}");
 
                 // Let the Session expire
-                await Task.Delay(TimeSpan.FromMinutes(1));
-
-                // Complete should throw
-                await Assert.ThrowsAsync<SessionLockLostException>(async () => await message.CompleteAsync());
-                try
-                {
-                    await messageSession.CloseAsync();
-                }
-                catch (Exception e)
-                {
-                    TestUtility.Log($"Got Exception on Session Close(): SessionId: {messageSession.SessionId}, Exception: {e.Message}");
-                }
-
-                messageSession = await queueClient.AcceptMessageSessionAsync(sessionId);
-            Assert.NotNull(messageSession);
-
-                message = await messageSession.ReceiveAsync();
-                TestUtility.Log($"Received Message: SessionId: {messageSession.SessionId}");
-
-                await message.CompleteAsync();
+                TestUtility.Log($"Waiting for session lock to time out...");
+                await Task.Delay(sessionReceiver.LockedUntilUtc - DateTime.UtcNow);
```

Comment:
Had added it initially and removed it purposely so we can see the failure, know why it happenned and then may be add the buffer. That is why Im also tracing the SessionLockedUntilUtc so we can see what is the issue, is it like a timelag or something else.